### DISCUSSION TITLE
spack-stack installation error on nautilus

### DISCUSSION THREAD

**Discussion by feiliuesmf** [JCSDA/spack-stack#889] [General]
Hi I am sorry I am not sure if there is a place to get some installation related help. I ran into an installation problem and it'd be great to get some help.

git clone --recurse-submodules https://github.com/JCSDA/spack-stack
. spack-stack/setup.sh
I would like to build with latest version of esmf (8.7.0b03) whose configuration is similar to 8.5.0

diff --git a/configs/common/packages.yaml b/configs/common/packages.yaml
index b5608ae..ee3909f 100644
--- a/configs/common/packages.yaml
+++ b/configs/common/packages.yaml
@@ -62,7 +62,7 @@
     # config and update the projections for lmod/tcl.
     # Also, check the acorn and derecho site configs which have esmf modifications.
     esmf:
-      version: ['8.5.0']
+      version: ['8.7.0b03']
       variants: ~xerces ~pnetcdf snapshot=none ~shared +external-parallelio
       require:
         - any_of: ['fflags="-fp-model precise" cxxflags="-fp-model precise"']

I added an entry for 8.7.0b03 in ./spack/var/spack/repos/builtin/packages/esmf/package.py
    # generate chksum with spack checksum esmf@x.y.z
    version("8.7.0b03", sha256="c8a40590e7d0d0123dfd54d48799a98b7ab4abde")
    version("8.5.0", sha256="acd0b2641587007cc3ca318427f47b9cae5bfd2da8d2a16ea778f637107c29c4")

Then I ran the following commands:
spack stack create env --site nautilus --template unified-dev --name espc_nautilus1
spack concretize 2>&1 | tee log.concretize
all ran successful up to this point.
Next, for installation
spack install --verbose --fail-fast 2>&1 | tee log.install

I ran into the following error which is related to wgrib2 package. Any tips on how to get the installation to work? Thanks,

/p/app/projects/espc/software/spackstack/spack-stack/spack/var/spack/repos/builtin/packages/wgrib2/package.py:158, in setup_build_environment:
        155        elif self.spec.compiler.name in ["gcc", "clang", "apple-clang"]:
        156            comp_sys = "gnu_linux"
        157
  >>    158        env.set("COMP_SYS", comp_sys)

==> Warning: Skipping build of ufs-srw-app-env-1.0.0-iagcc2uct4wvqyajm2klcza4mynlwnxc since wgrib2-3.1.1-crdeprl2flhdcuibla6cupscs6atk32p failed
==> Warning: Skipping build of global-workflow-env-1.0.0-y6sea7lejbzf3nnj6e5spl5zsox5mw62 since wgrib2-3.1.1-crdeprl2flhdcuibla6cupscs6atk32p failed
==> Error: Terminating after first install failure: UnboundLocalError: local variable 'comp_sys' referenced before assignment

**Comment by climbfuji** [comment]
Just to confirm, did you run the commands for Nautilus described in https://spack-stack.readthedocs.io/en/latest/PreConfiguredSites.html#navy-hpcmp-nautilus BEFORE running `source setup.sh` for spack-stack?

**Comment by feiliuesmf** [comment]
no, I did not. Restarting...

**Comment by feiliuesmf** [comment]
Now my build is hanging at this step (I restarted from scratch):
spack concretize 2>&1 | tee log.concretize

**Reply by climbfuji** [reply]
I am building on Nautilus now. The concretization step finished fairly quickly (1-2mins), install is running.

**Comment by feiliuesmf** [comment]
Thanks Dom, can you send me a list of the exact commands you use on nautilus so I can reproduce what you have? You can email me at fei.liu@gmail.com as well.

**Comment by feiliuesmf** [comment]
btw, it's stuck on python3 process acquiring a lock espc_nautilus2/.spack-env/transaction_lock. It looks like there may be some subtle race condition going on. I am not sure why.
